src/Domain/Repositories/ApprenticeshipQueryRepository.cs

```diff
@@ -10,302 +10,326 @@
-             ).ToList();
-         }
-         catch (Exception e)
-         {
-             _logger.LogError(e, "Error getting apprenticeships with episodes for provider UKPRN {Ukprn}", ukprn);
-         }
+        try
+        {
+            var apprenticeship = await DbContext.Apprenticeships
+                .Include(x => x.Episodes)
+                .Include(x => x.FreezeRequests)
+                .FirstOrDefaultAsync(x => x.Key == apprenticeshipKey);
+
+            var episodes = apprenticeship?.Episodes.ToList();
+            var latestEpisode = episodes?.MaxBy(x => x.Prices.Where(x => !x.IsDeleted).Select(x => x.StartDate));
+            if (latestEpisode == null)
+            {
+                return null;
+            }
+
+            paymentStatus = new PaymentStatus() { IsFrozen = latestEpisode.PaymentsFrozen };
+
+            if (paymentStatus.IsFrozen)
+            {
+                var activeFreezeRequest = apprenticeship!.FreezeRequests.Single(x => x.ApprenticeshipKey == apprenticeshipKey && !x.Unfrozen);
+                paymentStatus.Reason = activeFreezeRequest.Reason;
+                paymentStatus.FrozenOn = activeFreezeRequest.FrozenDateTime;
+            }
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error getting payment status for apprenticeship {apprenticeshipKey}", apprenticeshipKey);
+        }
+
+        return paymentStatus;
+    }
+
+    public async Task<List<ApprenticeshipWithEpisodes>?> GetApprenticeshipsWithEpisodes(long ukprn)
+    {
+        List<ApprenticeshipWithEpisodes>? apprenticeshipWithEpisodes = null;
+        try
+        {
+            var apprenticeships = await DbContext.Apprenticeships
+                .Include(x => x.Episodes)
+                .ThenInclude(x => x.Prices.Where(y => !y.IsDeleted))
+                .Where(x => x.Episodes.Any(e => e.Ukprn == ukprn))
+                .ToListAsync();
+
+            apprenticeshipWithEpisodes = apprenticeships.Select(apprenticeship =>
+                new ApprenticeshipWithEpisodes(
+                    apprenticeship.Key,
+                    apprenticeship.Uln,
+                    apprenticeship.GetStartDate(),
+                    apprenticeship.GetPlannedEndDate(),
+                    apprenticeship.Episodes.Select(ep =>
+                            new Episode(ep.Key, ep.TrainingCode, ep.Prices.Select(p =>
+                                new EpisodePrice(p.StartDate, p.EndDate, p.TrainingPrice, p.EndPointAssessmentPrice, p.TotalPrice, p.FundingBandMaximum)).ToList()))
+                        .ToList(),
+                    apprenticeship.GetAgeAtStartOfApprenticeship())
+            ).ToList();
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error getting apprenticeships with episodes for provider UKPRN {Ukprn}", ukprn);
+        }
 
         return apprenticeshipWithEpisodes;
-     }
- }
\ No newline at end of file
+    }
```

Comment:
The indentation in this file was a mess, I fixed it all which is why it's flagged a mass of changes above. Below this line is the method I added for this ticket.

Reply:
Fantastic, thank you